### PR TITLE
Apply advancements async

### PR DESCRIPTION
**Pull request by Proximyst**
This makes advancements load asynchronously upon login.

This does not optimise the speed at which they will load or be processed, but
will simply let the player move on in their login before they are all loaded.

### CODE REVIEW

File: Spigot-Server-Patches/0533-Apply-advancements-async.patch [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+ 
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+index dcbda5b35ab4d44e6b1d6c34bb8763a7dd08e629..af21c589c9c04d3df2506652a625f6f9761fdb85 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftPlayer.java
+@@ -1870,7 +1870,6 @@ public class CraftPlayer extends CraftHumanEntity implements Player {
+         CraftAdvancement craft = (CraftAdvancement) advancement;
+         AdvancementDataPlayer data = getHandle().getAdvancementData();
+         AdvancementProgress progress = data.getProgress(craft.getHandle());
+-
```
Comment (MiniDigger) [inline Spigot-Server-Patches/0533-Apply-advancements-async.patch]: whitespace!

**Comment by MiniDigger** [comment]
is it possible for a plugin to access advancements before they are loaded? if so, what happens?

**Comment by Proximyst** [comment]
They can start accessing it, but it will simply block until the advancements are loaded: https://github.com/PaperMC/Paper/pull/3454/files#diff-9ab7d96592886294eeb47a5c4530e55dR54

File: Spigot-Server-Patches/0533-Apply-advancements-async.patch (hunk 1 above)
Comment (Proximyst) [reply Spigot-Server-Patches/0533-Apply-advancements-async.patch]: Fixed :)

**Review by Toffikk — APPROVED** [review]
Any information on , when it's gonna be merged? , and , it works fine for me

**Comment by Proximyst** [comment]
After 1.16 most likely. We'll see :)

**Comment by Toffikk** [comment]
Could you update this PR? or is it abandoned?

**Comment by aikar** [comment]
I'm not sure if we need this now. We recently merged a PR that helped fix performance issue of loading advancements.

Would like to see some profiler results showing its still heavy on latest.

**Comment by Proximyst** [comment]
I'm closing this PR as if it is necessary, I want to reimplement it in another way to not break NMS plugins like Citizens.

**Pull request closed without merging**